lib/main.dart

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:js_interop';
+import 'dart:js_util';
+
+import 'package:path/path.dart' as path;
+import 'package:setup_dart/node/actions/tool_cache.dart';
+
+import 'node/actions/core.dart';
+import 'node/actions/exec.dart';
+import 'node/actions/http_client.dart';
+import 'node/fs.dart';
+import 'node/os.dart';
+import 'node/process.dart';
+
+void main(List<String> args) async {
+  try {
+    // sdk
+    var sdk = core.getInput('sdk');
+    if (sdk.isEmpty) {
+      sdk = 'stable';
+    }
+
+    // flavor
+    var flavor = core.getInput('flavor');
+    if (flavor.isEmpty) {
+      flavor = sdk == 'main' ? 'raw' : 'release';
+    } else if (flavor != 'raw' && flavor != 'release') {
+      core.setFailed("Unrecognized build flavor '$flavor'.");
+      return;
+    }
+    final raw = flavor == 'raw';
+
+    // os
+    final os = getPlatform();
+
+    // architecture
+    var architecture = core.getInput('architecture');
+    if (architecture.isEmpty) {
+      architecture = getArch();
+    }
+
+    // calculate version and channel
+    String version;
+    String channel;
+
```

Comment:
super nit: this newline seems inconsistent with the style above

Reply:
Hmm, I don't see anything different about it? Feel free to use the 'add a suggestion' feature to edit the file here.

Reply:
I meant to remove the newline, sorry for being unclear:

All the other blocks where:
<declaractions>
<if statement>

Except this which is:

<declarations>
<newline>
<if statement>

Not important though, so let's just resolve this.